exasol/analytics/audit/audit.py

```diff
@@ -124,41 +150,64 @@ def _count_rows(self, query: ModifyQuery, event_name: str) -> str:
         + ROW_COUNT: subquery
         + LOG_SPAN_NAME: query.db_operation_type: DbOperationType
         - LOG_SPAN_ID
-        - PARENT_LOG_SPAN_ID
+        + PARENT_LOG_SPAN_ID: query.parent_log_span.id
         + EVENT_NAME: parameter event_name
         - EVENT_ATTRIBUTES
-        + OBJECT_TYPE: query.db_object_type: DbObjectType
-        + OBJECT_SCHEMA: query.db_object_name: DBObjectName
-        + OBJECT_NAME: query.db_object_name: DBObjectName
+        + DB_OBJECT_TYPE: query.db_object_type: DbObjectType
+        + DB_OBJECT_SCHEMA: query.db_object_name: DBObjectName
+        + DB_OBJECT_NAME: query.db_object_name: DBObjectName
         - ERROR_MESSAGE
         """
 
+        db_obj = query.db_object_name
+        modify_query_fields = self._modify_query_fields(query, event_name)
+        other_table = query.db_object_name.fully_qualified
+        count_rows = base_column_values(
+            {BaseAuditColumns.ROW_COUNT: f"(SELECT count(1) FROM {other_table})"}
+        )
+        return self._insert_statement(
+            constants=(modify_query_fields | query.audit_fields),
+            scalar_functions=count_rows,
+        )
+
+    def _modify_query_fields(
+        self, query: ModifyQuery, event_name: str
+    ) -> dict[str, Any]:
         def schema(db_obj: DBObjectName) -> str | None:
             if not isinstance(db_obj, DBObjectNameWithSchema):
                 return None
             schema = cast(DBObjectNameWithSchema, db_obj).schema_name
             return schema.name if schema else None
 
         db_obj = query.db_object_name
-        query_attributes = {
-            "LOG_SPAN_NAME": query.db_operation_type.name,
-            "EVENT_NAME": event_name,
-            "DB_OBJECT_TYPE": query.db_object_type.name,
-            "DB_OBJECT_SCHEMA": schema(db_obj),
-            "DB_OBJECT_NAME": db_obj.name,
-        }
-        other_table = query.db_object_name.fully_qualified
-        row_count = BaseAuditColumns.ROW_COUNT.name.name
+        return base_column_values(
+            {
```

Comment:
Replaced literal columns names by references to `BaseAuditColumns` and using function `base_column_values()` to convert into a dict with strings as keys.